File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/relationship/v1_0/GoalCode.java
package com.evernym.verity.sdk.protocols.relationship.v1_0;

/**
 * An enumeration of possible goals (reasons) for a relationship.
 *
 * @see <a href="https://github.com/hyperledger/aries-rfcs/blob/master/features/0434-outofband/README.md" target="_blank" rel="noopener noreferrer">Aries 0434: Connection Protocol</a>
 */
public enum GoalCode {

    /**
     * To issue a credential
     */
    ISSUE_VC("issue-vc", "To issue a credential"),
    /**
     * To request a proof
     */
    REQUEST_PROOF("request-proof", "To request a proof"),
    /**
     * To create an account with a service
     */
    CREATE_ACCOUNT("create-account", "To create an account with a service"),
    /**
     * To establish a peer-to-peer messaging relationship
     */
    P2P_MESSAGING("p2p-messaging", "To establish a peer-to-peer messaging relationship");

    private final String code;
    private final String goalName;

    /**
     * Goal code is a tuple with a short code string and human readable name string
     * @param code short code string
     * @param name human readable string
     */
    GoalCode(String code, String name) {
        this.code = code;
        this.goalName = name;
    }

    /**
     * @return the defined short code string
     */
    public String code() {
        return code;
    }

    /**
     * @return @return the defined human readable name string
     */
    public String goalName() {
        return goalName;
    }

}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/Relationship/v1_0/GoalCode.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Json;
using System.Reflection;
using System.Runtime.CompilerServices;

namespace VeritySDK.Protocols.Relationship
{
    /// <summary>
    /// An enumeration of possible goals (reasons) for a relationship. 
    /// </summary>
    public enum GoalCode
    {
        /// <summary>
        /// To issue a credential
        /// </summary>
        [Display(Name = "issue-vc", Description = "To issue a credential")]
        ISSUE_VC,

        /// <summary>
        /// To request a proof
        /// </summary>
        [Display(Name = "request-proof", Description = "To request a proof")]
        REQUEST_PROOF,

        /// <summary>
        /// To create an account with a service
        /// </summary>
        [Display(Name = "create-account", Description = "To create an account with a service")]
        CREATE_ACCOUNT,

        /// <summary>
        /// To establish a peer-to-peer messaging relationship
        /// </summary>
        [Display(Name = "p2p-messaging", Description = "To establish a peer-to-peer messaging relationship")]
        P2P_MESSAGING
    }

    /// <summary>
    /// Extension for GoalCode enum
    /// </summary>
    public static class GoalCodeExtensions
    {
        /// <summary>
        /// Get Code for goal
        /// </summary>
        /// <param name="value">GoalCodeEnum</param>
        /// <returns></returns>
        public static string code(this GoalCode value)
        {
            return value.GetType()?
               .GetMember(value.ToString())?[0]?
               .GetCustomAttribute<DisplayAttribute>()?
               .Name;
        }

        /// <summary>
        /// Get Name for goal
        /// </summary>
        /// <param name="value">GoalCodeEnum</param>
        /// <returns></returns>
        public static string goalName(this GoalCode value)
        {
            return value.GetType()?
               .GetMember(value.ToString())?[0]?
               .GetCustomAttribute<DisplayAttribute>()?
               .Description;
        }
    }
}